manage-data/data-store/data-streams/failure-store.md

```diff
@@ -0,0 +1,2002 @@
+
+```console
+POST my-datastream-ingest/_doc
+{
+  "random": 42 // Not the field we're looking for.
+}
+```
+
+```console-result
+{
+  "_index": ".fs-my-datastream-ingest-2025.05.09-000002",
+  "_id": "eXS-tpYBwrYNjPmat9Cx",
+  "_version": 1,
+  "result": "created",
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "failed": 0
+  },
+  "_seq_no": 0,
+  "_primary_term": 1,
+  "failure_store": "used" // The document failed and went to the failure store.
+}
+```
+
+Inspecting the corresponding failure document will show the document in its original form as it was sent to {{es}}. 
+
+```console
+GET my-datastream-ingest::failures/_search
+```
+
+```console-result
+{
+  "took": 0,
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 1,
+      "relation": "eq"
+    },
+    "max_score": 1,
+    "hits": [
+      {
+        "_index": ".fs-my-datastream-ingest-2025.05.09-000002",
+        "_id": "eXS-tpYBwrYNjPmat9Cx",
+        "_score": 1,
+        "_source": {
+          "@timestamp": "2025-05-09T20:31:13.759Z",
+          "document": { <1>
+            "index": "my-datastream-ingest",
+            "source": {
+              "random": 42
+            }
+          },
+          "error": {
+            "type": "illegal_argument_exception",
+            "message": "field [data] not present as part of path [data]", <2>
+            "stack_trace": """j.l.IllegalArgumentException: field [data] not present as part of path [data]
+	at o.e.i.IngestDocument.getFieldValue(IngestDocument.java:202)
+	at o.e.i.c.SetProcessor.execute(SetProcessor.java:86)
+	... 14 more
+""",
+            "pipeline_trace": [
+              "my-datastream-example-pipeline"
+            ],
+            "pipeline": "my-datastream-example-pipeline",
+            "processor_type": "set"
+          }
+        }
+      }
+    ]
+  }
+}
+```
+1. The `document` field shows the state of the document is from before any pipeline executions.
+2. The pipeline failed after the timestamp would have been added.
+
+We can see that the document failed on the second processor in the pipeline. The first processor would have added a `@timestamp` field. Since the pipeline failed, we find that it has no `@timestamp` field added because it did not save any changes from before the pipeline failed.
+
+The second place failures can occur is during indexing. After the documents have been processed by any applicable pipelines, they are parsed using the index mappings before being indexed into the shard. If a document is sent to the failure store due to a failure in this process, then it will be stored as it was after any ingestion had occurred. This is because the original document is overwritten by the ingest pipeline changes by this point. This has the benefit of being able to see what the document looked like during the mapping and indexing phase of the write operation.
```

Comment:
```suggestion
The second time when failures can occur is during indexing. After the documents have been processed by any applicable pipelines, they are parsed using the index mappings before being indexed into the shard. If a document is sent to the failure store due to a failure in this process, then it will be stored as it was after any ingestion had occurred. This is because, by this point, the original document has already been overwritten by the ingest pipeline changes. This has the benefit of allowing you to see what the document looked like during the mapping and indexing phase of the write operation.
```
Small tweak.